client/material/left-nav/clickable-subheader.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Link, useRoute } from 'wouter'
+import { colorTextPrimary, colorTextSecondary } from '../../styles/colors'
+import { overline, singleLine } from '../../styles/typography'
+
+const Container = styled.div<{ $isActive: boolean }>`
+  width: 100%;
+  height: 36px;
+  background-color: ${props => (props.$isActive ? 'rgba(255, 255, 255, 0.08)' : 'transparent')};
+  color: ${props => (props.$isActive ? colorTextPrimary : colorTextSecondary)};
+  cursor: pointer;
+
+  &:hover {
+    background-color: rgba(255, 255, 255, 0.08);
+    color: ${colorTextPrimary};
+  }
+`
+
+const StyledLink = styled(Link)`
+  width: 100%;
+  height: 100%;
+  padding: 0 4px 0 16px;
+
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+
+  &:link,
+  &:visited,
+  &:hover,
+  &:active {
+    color: currentColor;
+    text-decoration: none;
+  }
+`
+
+const Title = styled.div`
+  ${overline};
+  ${singleLine};
+
+  height: 100%;
+  line-height: 36px;
+`
+
+const IconContainer = styled.div`
+  width: 36px;
+  height: 100%;
+
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`
+
+interface SubheaderProps {
+  icon?: React.ReactNode
+  children: React.ReactNode
+  className?: string
+}
+
+export const ClickableSubheader = React.forwardRef(
+  ({ icon, children, className }: SubheaderProps, ref: React.ForwardedRef<HTMLDivElement>) => {
+    const [isActive] = useRoute('/chat/list')
+
+    return (
```

Comment:
This still needs a ripple and such somewhere. Ideally the icon would also behave like an icon button (or just make it an icon button), just the click can be handled by this component

Reply:
Hmm, I'm not sure how to do this.

Just to recap, we want channel header to be a button (because that's the only way I can apply ripple effect to it afaik), with an icon button inside it (which has its own built-in ripple effect). Also, the whole thing needs to be inside a link as well?

Reply:
No, it is a link that is styled like a button. We want it to be a link so that search engine's can follow it (and because all the button did was navigate). We have plenty of other cases of "things that are not `<button>`s that use ripple, etc." for you to look at